Wikipedia/Code/WMFOnThisDayContentSource.m

```diff
@@ -120,24 +119,18 @@ - (void)loadContentForDate:(NSDate *)date inManagedObjectContext:(NSManagedObjec
 
                     [moc performBlock:^{
                         [onThisDayEvents enumerateObjectsUsingBlock:^(WMFFeedOnThisDayEvent *_Nonnull event, NSUInteger idx, BOOL *_Nonnull stop) {
-                            __block NSInteger countOfImages = 0;
                             [event.articlePreviews enumerateObjectsUsingBlock:^(WMFFeedArticlePreview *_Nonnull articlePreview, NSUInteger idx, BOOL *_Nonnull stop) {
-                                if (articlePreview.imageURLString) {
-                                    countOfImages += 1;
-                                }
                                 [moc fetchOrCreateArticleWithURL:[articlePreview articleURL] updatedWithFeedPreview:articlePreview pageViews:nil];
                             }];
-                            event.score = @(countOfImages);
+                            event.score = [event calculateScore];
                             event.index = @(idx);
                         }];
 
                         NSInteger featuredEventIndex = NSNotFound;
 
                         NSArray *eventsSortedByScore = [onThisDayEvents sortedArrayUsingDescriptors:@[[NSSortDescriptor sortDescriptorWithKey:@"score" ascending:NO]]];
                         if (eventsSortedByScore.count > 0) {
-                            NSInteger index = ((year % 10) % eventsSortedByScore.count);
-                            WMFFeedOnThisDayEvent *featuredEvent = eventsSortedByScore[index];
-                            featuredEventIndex = featuredEvent.index.integerValue;
+                            featuredEventIndex = ((WMFFeedOnThisDayEvent*)eventsSortedByScore.firstObject).index.integerValue;
```

Comment:
Removing the `year % 10` means you'll likely repeat the same events every year

Reply:
Assuming the event text and article previews image counts remained static over the course of a year that would definitely be true. Not sure it's likely, but I can try to make the score use some weight based on current year... will have to think a bit about how that will play with the current mechanism... will take a stab at it :)

Reply:
@joewalsh 

So if I just leave this as it was...

     NSInteger index = ((year % 10) % eventsSortedByScore.count);
     WMFFeedOnThisDayEvent *featuredEvent = eventsSortedByScore[index];		
     featuredEventIndex = featuredEvent.index.integerValue;

... it seems like it would be ok. Guessing that's better than moving any current year logic into `calculateScore`?